src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -294,603 +363,1248 @@ public static boolean hasToolLevel(final String tool, final ItemStack stack, fin
+    public static int getItemCountInProvider(@NotNull final ICapabilityProvider provider, @NotNull final Item targetItem, int itemDamage)
     {
-        for (int i = 0; i < inventory.getSizeInventory(); i++)
-        {
-            final ItemStack item = inventory.getStackInSlot(i);
-            //Only classic fishingRod recognized as a fishingTool
-            if (item != null && (item.getItem().getToolClasses(item).contains(tool) || ("hoe".equals(tool) && item.getUnlocalizedName().contains("hoe"))
-                                   || ("rod".equals(tool) && item.getUnlocalizedName().contains("fishingRod"))))
-            {
-                return i;
-            }
-        }
-        return -1;
+        return getItemCountInProvider(provider, (ItemStack stack) -> compareItems(stack, targetItem, itemDamage));
     }
 
     /**
-     * Verifies if there is one tool with an acceptable level
-     * in a worker's inventory.
+     * Returns the amount of occurrences in the {@link ICapabilityProvider}.
      *
-     * @param tool      the type of tool needed
-     * @param inventory the worker's inventory
-     * @param hutLevel  the worker's hut level
-     * @return true if tool is acceptable
+     * @param provider                    {@link ICapabilityProvider} to scan.
+     * @param itemStackSelectionPredicate The predicate used to select the
+     *                                    stacks to count.
+     * @return Amount of occurrences of stacks that match the given predicate.
      */
-    public static boolean hasToolLevel(final String tool, @NotNull final IInventory inventory, final int hutLevel)
+    public static int getItemCountInProvider(@NotNull final ICapabilityProvider provider, @NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
     {
-        for (int i = 0; i < inventory.getSizeInventory(); i++)
-        {
-            final ItemStack item = inventory.getStackInSlot(i);
-            final int level = Utils.getMiningLevel(item, tool);
+        return getItemHandlersFromProvider(provider).stream()
+                 .mapToInt(handler -> filterItemHandler(handler, itemStackSelectionPredicate).stream().mapToInt(InventoryUtils::getItemStackSize).sum())
+                 .sum();
+    }
 
-            if (Utils.isTool(item, tool) && verifyToolLevel(item, level, hutLevel))
-            {
-                return true;
-            }
-        }
-        return false;
+    /**
+     * Checks if a player has a block in the {@link ICapabilityProvider}.
+     * Checked by {@link #getItemCountInProvider(ICapabilityProvider, Block,
+     * int)} &gt; 0;
+     *
+     * @param Provider   {@link ICapabilityProvider} to scan
+     * @param block      Block to count
+     * @param itemDamage the damage value.
+     * @return True when in {@link ICapabilityProvider}, otherwise false
+     */
+    public static boolean hasItemInProvider(@NotNull final ICapabilityProvider Provider, @NotNull final Block block, int itemDamage)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)